data/meterpreter/meterpreter.py

```diff
@@ -479,7 +479,7 @@ def get_packet_tcp(self):
 			pkt_length -= 8
 			packet = bytes()
 			while len(packet) < pkt_length:
-				packet += self.socket.recv(4096)
+				packet += self.socket.recv(pkt_length - len(packet))
```

Comment:
I don't see why not, however calling len only on new data each time like you did is more efficient.